code/iaas/logic/src/main/resources/META-INF/cattle/core-process/spring-core-processes-context.xml

```diff
@@ -144,4 +144,25 @@
     <!-- Cluster host map -->
     <process:process name="clusterhostmap.create" resourceType="clusterHostMap" start="requested" transitioning="creating" done="created" />
     <process:process name="clusterhostmap.remove" resourceType="clusterHostMap" start="created,creating" transitioning="removing" done="removed" />
+
+    <!-- Service Discovery Environment -->
+    <process:process name="environment.create" resourceType="environment" start="requested" transitioning="activating" done="active" />
+    <process:process name="environment.activateservices" resourceType="environment" start="active" transitioning="updating-active" done="active" />
+    <process:process name="environment.deactivateservices" resourceType="environment" start="active" transitioning="updating-active" done="active" />
+    <process:process name="environment.remove" resourceType="environment" start="active, activating, updating-active" transitioning="removing" done="removed" />
+    
+    <!-- Service Discovery Service -->
+    <process:process name="service.create" resourceType="service" start="requested" transitioning="registering" done="inactive" />
+    <process:process name="service.activate" resourceType="service" start="inactive" transitioning="activating" done="active" />
+    <process:process name="service.deactivate" resourceType="service" start="active,activating,updating-active" transitioning="deactivating" done="inactive" />
```

Comment:
@sonchang Good catch; i used to have the process setting the state to updating-active, then I remove it, but forgot to remove it from the start state for the object.